### DISCUSSION TITLE
Implementation of switch function feature to allow for conditional branching

### DISCUSSION THREAD

**Discussion by vzhang03** [jspsych/jsPsych#3323] [Ideas]
Building upon [previous discussion](https://github.com/jspsych/jsPsych/issues/314), to address the issues of conditional branching but maintaining syntax close to existing if-statements for simplicity, I would like to propose the syntax below:

```
var switch_node = {
    timeline: 
       {
            criminal: criminal_timeline, 
            healthcare: healthcare_timeline, 
            education: education_timeline,
            vaccine: vaccine_timeline
       },
    switch_function: function(){
        var data = jsPsych.data.get().last(1).values()[0];
        switch (jsPsych.pluginAPI.compareKeys(data.response)){
            case criminal: 
               return 'criminal';
            case healthcare: 
               return 'healthcare';
            case education: 
               return 'education';
            case vaccine: 
               return 'vaccine';
            default:   
               return 'else'; // skip? or no default
    }
}
```

This format maintains the same syntax as [normal conditional branching,](https://www.jspsych.org/v7/overview/timeline/) except the main difference is a dictionary to organize responses and string return values. Compared to normal conditional branching where the return values 'true' or 'false' map to whether the timeline happens, having flexible return values of string keys allows us to account for more than two branches. Having these string keys map to different timelines defined by representing timeline_variables as an object allows this to be clear and efficient in use and implementation.

**Comment by jodeleeuw** [comment]
@jspsych/core would love to get some thoughts on this!

**Comment by jodeleeuw** [comment]
Another option that we could pursue would be to allow `timeline` to be a function that returns the timeline to run. Then it would look like this:

```js
var switch_node = {
    timeline: function(){
        var data = jsPsych.data.get().last(1).values()[0];
        switch (jsPsych.pluginAPI.compareKeys(data.response)){
            case criminal: 
               return criminal_timeline;
            case healthcare: 
               return healthcare_timeline;
            case education: 
               return education_timeline;
            case vaccine: 
               return vaccine_timeline;
            default:   
               return 'else'; // skip? or no default
    }
}
```
